### PR TITLE
Mastermode is stored as a configuration entry

### PR DESCRIPTION
**Pull request by francinum**
## Changelog
:cl:
config: Default gamemode is now a config entry.
/:cl:

### CODE REVIEW

**Review by PowerfulBacon — DISMISSED** [review]

**Comment by jlsnow301** [comment]
~~very fixable conflict with my PR. i can update once this passes.~~ (EDIT: Done) also, this PR fixes the "round restart vote passed" message on game mode vote

**Comment by francinum** [comment]
I still have no idea why the test are failing.

**Comment by Crossedfall** [comment]
Re-running tests, mostly because I don't remember what the error was and all the old ones were listed as canceled.

Edit: Ah, right. It never started after initialization 

Edit 2: Still unable to determine the cause. I've created a linux hosted docker container that runs through the CI tests and it was able to start and get through the tests just fine.

File: code/_globalvars/game_modes.dm [hunk 1]
```diff
@@ -1,4 +1,4 @@
-GLOBAL_VAR_INIT(master_mode, "traitor") //! "extended"
+GLOBAL_VAR_INIT(master_mode, "secret") //! Stores the current gamemode. Fuck you CIscript.
```
Comment (Crossedfall) [inline code/_globalvars/game_modes.dm]:
```suggestion
GLOBAL_VAR_INIT(master_mode, "extended") //! Stores the current gamemode. Fuck you CIscript.
```

File: code/controllers/configuration/entries/game_options.dm [hunk 2]
```diff
@@ -450,3 +450,7 @@
 /datum/config_entry/flag/restricted_suicide
 
 /datum/config_entry/flag/dynamic_config_enabled
+
+// Master Mode
+/datum/config_entry/string/master_mode
+	config_entry_value = "secret"
```
Comment (Crossedfall) [inline code/controllers/configuration/entries/game_options.dm]:
```suggestion
	config_entry_value = "extended"
```

**Review by Crossedfall — CHANGES_REQUESTED** [review]
This wont pass the CI due to lacking configs. The CI runs with an absolutely minimum config file, whereas this requires secret config options to function. Defaulting to `extended` resolves this

Edit: This is the full config for the CI https://github.com/BeeStation/BeeStation-Hornet/blob/master/tools/ci/ci_config.txt

File: code/_globalvars/game_modes.dm [hunk 3]
```diff
@@ -1,4 +1,4 @@
-GLOBAL_VAR_INIT(master_mode, "traitor") //! "extended"
+GLOBAL_VAR_INIT(master_mode, "extended") //! Stores the current gamemode. Fuck you CIscript.
```
Comment (francinum) [inline code/_globalvars/game_modes.dm]:
```suggestion
GLOBAL_VAR_INIT(master_mode, "Loading...") //! Stores the current gamemode. Intentionally invalid placeholder is replaced in Ticker.
```

**Comment by ike709** [comment]
@Crossedfall config

**Review by Crossedfall — APPROVED** [review]
Config applied

**Pull request merged**